internal/cli/kraft/cloud/scale/initialize/initialize.go

```diff
@@ -109,80 +109,91 @@ func (opts *InitOptions) Run(ctx context.Context, args []string) error {
 		return fmt.Errorf("cooldown time must be at least 1ms")
 	}
 
-	master := &kraftcloudinstances.Instance{}
+	var master kcautoscale.CreateRequestMaster
 
 	if opts.Master == "" {
 		if config.G[config.KraftKit](ctx).NoPrompt {
 			return fmt.Errorf("specify an instance master UUID or name via --master")
 		}
 
-		instances, err := opts.Client.Instances().WithMetro(opts.Metro).List(ctx)
+		instListResp, err := opts.Client.Instances().WithMetro(opts.Metro).List(ctx)
 		if err != nil {
 			return fmt.Errorf("could not list instances: %w", err)
 		}
-
-		if len(instances) == 0 {
-			return fmt.Errorf("no instances found in service group")
+		if len(instListResp) == 0 {
+			return fmt.Errorf("no instance found in service group")
 		}
 
-		if len(instances) == 1 {
-			master.UUID = instances[0].UUID
+		if len(instListResp) == 1 {
+			master.UUID = &instListResp[0].UUID
 		} else {
-			var possible []kraftcloudinstances.Instance
+			var possible []stringerInstance
 
-			for _, instance := range instances {
-				if instance.ServiceGroup == nil {
+			for _, instItem := range instListResp {
+				i, err := opts.Client.Instances().WithMetro(opts.Metro).GetByUUID(ctx, instItem.UUID)
+				if err != nil {
+					return fmt.Errorf("could not get details of  instances %s: %w", instItem.UUID, err)
+				}
+				if i.ServiceGroup == nil {
 					continue
 				}
-
-				if instance.ServiceGroup.UUID != args[0] {
+				if i.ServiceGroup.UUID != args[0] {
 					continue
 				}
 
-				possible = append(possible, instance)
+				possible = append(possible, stringerInstance{i})
```

Comment:
This got me properly confused, what Go magic is this?

Not even sure what a `stringerInstance` is

Reply:
This type is at the bottom of the file.
KraftKit requires types to be `fmt.Stringer` to be able to perform the interactive selection, but the SDK has no reason to expose stringer types (what does it mean for an API response to be `fmt.Stringer`, what does it print? All fields, some fields?).

Since Go interfaces can be implemented close to where they are actually required, we simply use a mini wrapper `fmt.Stringer` wrapper type around the API response, and voilà: KraftKit works without touching the SDK.